Templates/colors.json

```diff
@@ -0,0 +1,4687 @@
+        { "name": "IsReset", "type": "bool" },
+        { "name": "IsLightSpeedDashTarget", "type": "bool" },
+        { "name": "SetPlaceType", "type": "ObjRingSpawner::SetPlaceType" },
+        { "name": "EventType", "type": "ObjRingSpawner::EventType" },
+        { "name": "RotType", "type": "ObjRingSpawner::RotType" }
+      ]
+    },
+    "ObjSuperRingSpawner": {
+      "fields": [
+        { "name": "TargetID", "type": "object_reference" },
+        { "name": "SetPlaceType", "type": "ObjSuperRingSpawner::SetPlaceType" },
+        { "name": "EventType", "type": "ObjSuperRingSpawner::EventType" }
+      ]
+    },
+    "ObjBombBlockSpawner": {
+      "fields": [
+        { "name": "Counter", "type": "uint32" }
+      ]
+    },
+    "ObjPlasmaCannonSpawner": {
+      "fields": [
+        { "name": "ShotSpeed", "type": "float32" },
+        { "name": "ShotInterval", "type": "float32" },
+        { "name": "BarrageInterval", "type": "float32" },
+        { "name": "ShotLifeTime", "type": "float32" },
+        { "name": "BarrageType", "type": "ObjPlasmaCannonSpawner::BarrageType" },
+        { "name": "ActionType", "type": "ObjPlasmaCannonSpawner::ActionType" }
+      ]
+    },
+    "ObjReproductionSpawner": {
+      "fields": [
+        { "name": "ReviveTime", "type": "float32" },
+        { "name": "Type", "type": "ObjReproductionSpawner::Type" }
+      ]
+    },
+    "ObjShootingSpawner": {
+      "fields": [
+        { "name": "MoveRate", "type": "float32" },
+        { "name": "MoveLength", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "MoveType", "type": "ObjShootingSpawner::MoveType" },
+        { "name": "ActionType", "type": "ObjShootingSpawner::ActionType" }
+      ]
+    },
+    "ObjVaultSpawner": {
+      "fields": [
+        { "name": "MoveSpeed", "type": "float32" },
+        { "name": "GuidePathID", "type": "uint32" },
+        { "name": "Path1DPos", "type": "float32" },
+        { "name": "ActionType", "type": "ObjVaultSpawner::ActionType" },
+        { "name": "OffsetType", "type": "ObjVaultSpawner::OffsetType" }
+      ]
+    },
+    "ObjBigCatapultSpawner": {
+      "fields": [
+        { "name": "UpSpeed", "type": "float32" },
+        { "name": "DownSpeed", "type": "float32" },
+        { "name": "Height", "type": "float32" },
+        { "name": "StopTime", "type": "float32" },
+        { "name": "DeadPos", "type": "float32" },
+        { "name": "CollisionWidth", "type": "float32" },
+        { "name": "Size", "type": "ObjBigCatapultSpawner::Size" },
+        { "name": "Mode", "type": "ObjBigCatapultSpawner::Mode" },
+        { "name": "IsDeadTypeCrush", "type": "bool" },
+        { "name": "IsMessageBoot", "type": "bool" }
+      ]
+    },
+    "ObjBilliardSpawner": {
+      "fields": [
+        { "name": "ModelType", "type": "ObjBilliardSpawner::ModelType" },
+        { "name": "NumMaxCollisions", "type": "uint8" }
+      ]
+    },
```

Comment:
Parameters named based on in-game functionality and parameter names from other objects with similar functionality.